flink-ml-lib/src/main/java/org/apache/flink/ml/common/util/QuantileSummary.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.util;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Helper class to compute an approximate quantile summary. This implementation is based on the
+ * algorithm proposed in the paper: "Space-efficient Online Computation of Quantile Summaries" by
+ * Greenwald, Michael and Khanna, Sanjeev. (https://doi.org/10.1145/375663.375670)
+ */
+@Internal
```

Comment:
According to the Java doc of `@Internal`, it is an "Annotation to mark methods within stable, public APIs as an internal developer API".

Thus it seems unnecessary to add this annotation.

@zhipeng93 Could you confirm that we should remove this annotation?

Reply:
It is a developper API and is expected to be used in different algorithms. 

It is some how similar as we marked `withBroadcast` as `Internal`. [1]

[1] https://github.com/apache/flink-ml/commit/2edf453251c7ed9d8d1778c9c0115194e3d13ca7

Reply:
@jiangxin369 If you need to update this PR, it might be simpler to also update `withBroadcast()` to follow the best practice of using `@Internal`.

Reply:
Updated.